core/templates/services/context.service.spec.ts

```diff
@@ -64,6 +64,24 @@ describe('Context service', () => {
       expect(ecs.getQuestionPlayerIsManuallySet()).toEqual(false);
       expect(ecs.isInQuestionPlayerMode()).toEqual(false);
     });
+
+    it('should affirm the page context to be exploration player or editor',
+      ()=> {
+        expect(ecs.isInExplorationContext()).toBe(true);
```

Comment:
Where are we setting the context in this test to be either player / editor?

Reply:
isInExplorationContext() returns true when getPageContext() is either exploration player or editor. `spyOn(urlService, 'getPathname').and.returnValue('/explore/123');` in beforeEach sets the pageContext to exploration player.

Reply:
We should be setting the context to either editor or player in this test and then expecting that the `isInExplorationContext` returns the correct value. Include a pre-check as well.

Reply:
BTW is this a duplicate? I see the same test in Lines 228 (player) and 235 (editor).

Reply:
Removed. Done!